hw/hw17/html_render.py

```diff
@@ -5,7 +5,109 @@
 
 """
 
+
 # The start of it all:
 # Fill it all in here.
 class Element(object):
-    pass
+
+    BASE_INDENT = "    "
+
+    def __init__(self, tag="", content=None, oneLineFlag=False,
+                 selfCloseFlag=False, **kwargs):
+        self.tag = tag
+        self.content = []
+        self.append(content)
+        self.oneLineFlag = oneLineFlag
+        self.selfCloseFlag = selfCloseFlag
+        self.attrs = []
+        self.attrStr = ''
+        if (kwargs is not None):
+            for key, val in kwargs.items():
+                self.attrs.append("%s=\"%s\"" % (key, val))
+        if (len(self.attrs) > 0):
+            self.attrStr = " " + " ".join(self.attrs)
+
+    def append(self, content):
+        if (content):
+            self.content.append(content)
+
+    def render(self, fh, indent=""):
+        if self.selfCloseFlag:
+            fh.write("%s<%s%s />\n" % (indent, self.tag, self.attrStr))
+        else:
+            fh.write("%s<%s%s>" % (indent, self.tag, self.attrStr))
+            if not self.oneLineFlag:
+                fh.write("\n")
+            curIndent = indent + Element.BASE_INDENT
+            for content in self.content:
+                if (isinstance(content, str)):
+                    if self.oneLineFlag:
+                        fh.write("%s" % (content))
+                    else:
+                        fh.write("%s%s\n" % (curIndent, content))
+                elif (isinstance(content, Element)):
+                    content.render(fh, curIndent)
+            if not self.oneLineFlag:
+                fh.write("%s" % (indent))
+            fh.write("</%s>\n" % (self.tag))
+
+
+class Html(Element):
+
+    def __init__(self, content="", **kwargs):
```

Comment:
My approach here was to have the tag as a class attribute rather than an argument. So just set tag='html' and there's no need to override the init every time.